Add OOB versioning suspended tests

welansari opened the pull request:
Issue: ZENKO-4330

Commit:
add versioning supended OOB tests

Issue: ZENKO-4330

Commit:
add test for ingesting non versioned objets

Issue: ZENKO-4330

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented on tests/zenko_tests/node_tests/backbeat/tests/ingestion/ingestionS3C.js:
```diff
@@ -9,27 +9,41 @@ const scalityUtils = new IngestionUtility(scalityS3Client, ringS3Client);
-    ], done));
+    afterEach(function (done) {
+        async.series([
+            next => ringS3CUtils.deleteAllVersions(
```
even if the location is created before, since we delete here we should also create the objects here : so the test may be (re)played multiple times...

welansari replied:
Putting the objects is not something we can do in this case. The whole reason behind putting the objects before the location gets created is to have a specific type of null version that doesn't have an internal version id, this type of object can only be created when the bucket is non versioned which cannot happen once versioning gets enabled on it.

I want to avoid having to create the location mid tests for flakiness reasons which is why i went with this approach.

francoisferrand replied:
in that case we should not delete the objects either : so that we can run the test again...

that said, location creation is not flaky as far as I know : this was done to test cold location creation, and the flakiness issue was dealt with (i think)... So we now (in ctst tests) how to do this...

williamlardier commented on .github/workflows/end2end.yaml:
```diff
@@ -81,6 +81,9 @@ env:
   RING_S3C_BACKEND_SOURCE_LOCATION: rings3cbackendingestion
   RING_S3C_INGESTION_SRC_BUCKET_NAME: ingestion-test-src-bucket-${{ github.run_id }}
+  RING_S3C_BACKEND_SOURCE_NON_VERSIONED_LOCATION: rings3cbackendingestionnonversioned
+  RING_S3C_INGESTION_SRC_NON_VERSIONED_BUCKET_NAME: ingestion-test-src-non-versioned-bucket-${{ github.run_id }}
```
the github run_id is an incrementing number (currently 11 char), meaning the bucket name length here is already 52 characters. To be future proof, maybe good to ensure the length cannot exceed 63 chars (the max for an s3 bucket name)

Commit:
remove github run_id from s3c bucket name

Since S3C (Metadata+Cloudserver) is deployed
locally on each run, there is no longer a need
to have a unique bucket name.

Previously, the tests used to target a persistent
RING deployed externally, hence the need of having
a unique bucket name for each run and attempt.

Issue: ZENKO-4330

welansari commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.10`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue ZENKO-4330.

Goodbye kerkesni.




*The following options are set:* **approve**
